### PR TITLE
Refactor and Simplify Main Unit Tests

### LINKED ISSUE #179: Refactor main unit-tests by leveraging testing.F.Cleanup
As mentioned in #169, as an alternative to having a testify suite which sets up all the things, we should consider making smaller, more targeted functions which use [testing.F.Cleanup](https://pkg.go.dev/testing#F.Cleanup) to clean up on test completion.

e.g.

```go
func newMockLambdaServer(t testing.TB) *http.Server {
    lambdaServer := ...
    t.Cleanup(lambdaServer.Close)
    return lambdaServer
}

func newMockAPMServer(t testing.TB) (*http.Server, *apmServerInternals) {
    ...
}
```

### PR DESCRIPTION
**Pull request by jlvoiseux**
### Motivation / Summary
This PR resolves #179, and contains changes that were taken out of #191 for focus purposes.

### Changes
24a5a89 remove `testify.Suite` in favor of self-sufficient tests, whose teardown is handled by `T.Cleanup()`
7129e46 uses `assert.Contains` as intended by remove useless boolean assertions

Older commits are irrelevant, as they were merged as part of #191

### CODE REVIEW

File: apm-lambda-extension/main_test.go [hunk 1]
```diff
@@ -199,10 +202,26 @@ func initMockServers(eventsChannel chan MockEvent) (*httptest.Server, *httptest.
...
+
+	t.Cleanup(func() { lambdaServer.Close() })
+	return &lambdaServerInternals
+}
+
+// TODO : Move logger out of extension package and stop using it as a package-level variable
+func newLogger(t *testing.T, logLevel string) {
+	if err := os.Setenv("ELASTIC_APM_LOG_LEVEL", logLevel); err != nil {
+		t.Fail()
 	}
```
Comment (axw) [inline apm-lambda-extension/main_test.go:216]:
```suggestion
	t.Setenv("ELASTIC_APM_LOG_LEVEL", logLevel)
```

File: apm-lambda-extension/main_test.go [hunk 2]
```diff
@@ -199,10 +202,26 @@ func initMockServers(eventsChannel chan MockEvent) (*httptest.Server, *httptest.
...
+}
 
-	return lambdaServer, apmServer, &apmServerInternals, &lambdaServerInternals
+func newTestStructs(t *testing.T) (context.Context, chan MockEvent) {
```
Comment (axw) [inline apm-lambda-extension/main_test.go:219]: AFAICS, the returned context is never used in any calls to `newTestStructs`. Can we get rid of it?

**Review by axw — APPROVED** [review]
LGTM overall, just a few other small things which I think we can tidy up. Generally, we should probably be using `T.Setenv` instead of `os.Setenv` directly.

File: apm-lambda-extension/main_test.go [hunk 3]
```diff
@@ -199,10 +202,26 @@ func initMockServers(eventsChannel chan MockEvent) (*httptest.Server, *httptest.
...
-		return nil, nil, nil, nil
+		return nil
+	}
+
+	t.Cleanup(func() { lambdaServer.Close() })
+	return &lambdaServerInternals
+}
+
+// TODO : Move logger out of extension package and stop using it as a package-level variable
+func newLogger(t *testing.T, logLevel string) {
```
[the remark this answers is not in the document]
Comment (jlvoiseux) [reply apm-lambda-extension/main_test.go]: Fixed in ecdc5f0

File: apm-lambda-extension/main_test.go (hunk 2 above)
Comment (jlvoiseux) [reply apm-lambda-extension/main_test.go]: Fixed in ecdc5f0

File: apm-lambda-extension/main_test.go [hunk 4]
```diff
@@ -83,9 +83,7 @@ type ApmInfo struct {
 	Version      string    `json:"version"`
 }
 
-func initMockServers(eventsChannel chan MockEvent) (*httptest.Server, *httptest.Server, *MockServerInternals, *MockServerInternals) {
-
-	// Mock APM Server
+func newMockApmServer(t *testing.T) (*MockServerInternals, *httptest.Server) {
```
[the remark this answers is not in the document]
Comment (jlvoiseux) [reply apm-lambda-extension/main_test.go:86]: It is [required](https://github.com/elastic/apm-aws-lambda/blob/ecdc5f0471ef637b2e11e2cf9bf0bb4f07a2b40f/apm-lambda-extension/main_test.go#L412) in `TestAPMServerDown` were the Server is closed preemptively to simulate the scenario where the latter is down.

**Comment by jlvoiseux** [comment]
Thanks for the review @axw ! I have implemented your recommendations in ecdc5f0.

File: apm-lambda-extension/main_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (axw) [reply apm-lambda-extension/main_test.go:86]: I see, sorry - missed that in my search.

**Review by axw — APPROVED** [review]
LGTM, thanks :)

**Pull request merged**